### PR TITLE
feat: enable efb

### LINKED ISSUE #1251: [REQUEST] Electronic Flight Bag
**Is your feature request related to a problem? Please describe.**

no

**Describe the feature you'd like to see implemented**

A tablet in the tablet holder like avitab for xplane 11

**References**

has holder for ipad

**Additional context**

Discord username (if different from GitHub):

### PR DESCRIPTION
**Pull request by pepperoni505**
Fixes #1251

## Summary of Changes

* Enabled React EFB app to display in production builds

### Fixes

* [x] Status bar collapsing on big pages
* [x] Map shows wrong location (API issue - needs changes there)

## Testing instructions

* Make sure the screen is properly usable **(Functionality is almost non-existent at the moment)**
* Make sure there are no visual glitches
* Make sure the tablet turns on with aircraft AC power

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

File: src/instruments/src/EFB/index.jsx [hunk 1]
```diff
@@ -18,9 +18,6 @@
...
 import ReactDOM from 'react-dom';
 import { renderTarget } from '../util.mjs';
+import Efb from './Efb.tsx';
 
-const Efb = () => (
-    <div />
-);
-
-ReactDOM.render(<Efb />, renderTarget);
+ReactDOM.render(<Efb currentFlight="TS516" />, renderTarget);
```
Comment (tracernz) [inline src/instruments/src/EFB/index.jsx:23]: TS516 intentional?

Comment (ZigTag) [reply src/instruments/src/EFB/index.jsx:23]: It looks like some part of the shim code must have made its way to production

Comment (myselfaaryan) [reply src/instruments/src/EFB/index.jsx:23]: It should pass the ATC FLIGHT NUMBER SimVar

**Review by Benjozork — APPROVED** [review]
LGTM

**Review by myselfaaryan — APPROVED** [review]
LGTM!

**Review by ZigTag — DISMISSED** [review]
LGTM but can we disable the ground page until the buttons are functional?

File: src/instruments/src/EFB/index.jsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import ReactDOM from 'react-dom';
+import { useState } from 'react';
+import Efb from './Efb.tsx';
+import { renderTarget, useUpdate, getSimVar } from '../util.mjs';
+import logo from './Assets/fbw-logo.svg';
+
+import './Assets/Boot.scss';
+
+// TODO: Move anything dependent on ac power change to A32NX_Core
+function powerAvailable() {
```
Comment (davidwalschots) [inline src/instruments/src/EFB/index.jsx:28]: Just a silly question... Why does the iPad-like device care about the aircraft's power being available? Does it not have its own little battery?

Comment (Benjozork) [reply src/instruments/src/EFB/index.jsx:28]: It will have that logic later, the aircraft power thing is temporary

**Comment by BritishAvgeek** [comment]
Category: QA Tester
Name: BritishAvgeek#1436
Date of testing: 31/01/2021
Version of the sim: 1.12.13.0
PR Tested: #3252 
Tier of testing: 1
Changes to observe: EFB is alive

Testing technique: Spawned and gave the EFB the A320 version of the "kiss of life"
Steps you've made to test the Pull Request:
1.  Spawned C&D
2.  Ext Power On, Batt 1 and 2 On - EFB turned on
3.  Clicked through all the pages, pulled a Simbrief plan in after adding my simbrief username into the Company tab
4.  Checked the loadsheet and clicked about the entire EFB in it's current state

Media: https://youtu.be/5QyidfF4rR0

Issues (If Any): Map stuck me in USA... and the keybinds were causing me an issue as if I scrolled down on the loadsheet it also zoomed out so a lock out of some sort if possible would be cool (if not, i'll just change my keybinds for zoom though!)

Overall Rating: Good with future tweaks as it develops
Conclusions of the testing: It lives!!! I love it - the Dashboard is REALLY nice and I love the simbrief integration with the loadsheet too!

The proper METAR is a little small on the dashboard however - which is the one I'd read personally, though the "decoded METAR" being so clear is REALLY nice for the wider user experience and think it looks exceptionally good.

Keep up the great work as always FBW peoples! Love it.

**Comment by Awemeter** [comment]
Category: QA Tester
Name: Awemeter#7919 
Date of testing: 1/31/2020
Version of the sim: 1.12.13.0
PR Tested: #3252 
Tier of testing: 1
Changes to observe:
Enable EFB

Testing technique: Tried out
Steps you've made to test the Pull Request:
1. Spawned in C&D at KLAS from flytampa without EFB with batteries and external power on. FPS=41-44
2. Spawned in C&D at KLAS from flytampa with EFB with batteries and external power on. FPS=33-45
3. Checked EFB functionality like pulling simbreif report and clicking through the load sheet page using the map.


Flight notes (When done a full flight):
2080Ti
I9-9900k
64GB of ram
running all ultra settings.
Media: Insert screenshots, recordings, or anything relevant to the review here.
image

Issues (If Any): Map showing the wrong location
image



Suggestions:
Cancel scrolling when on EFB
Change Battery icon to White or Green
Change default position on the map
Add a button that will bring you back to your plane on the map
Add a dark mode icon 
image
Add clicking effect like on your phone when you click an app the app becomes dark reacting to your click
Change from using wifi to cell service since that is more common when on the ground.

Overall Rating: To be improved
Conclusions of the testing: A very epic PR and looks amazing I can't wait for more features to come. Thank you to all the epic people that worked not the efb from getting the model into the plane to making the amazing UI.

**Comment by oliverpope03** [comment]
Just doing some private testing sat on the ground at EGNX waiting for departure and got this lovely thing https://gyazo.com/b446bd9275455aa714961706a46ddda0

**Comment by Benjozork** [comment]
> 
> 
> Just doing some private testing sat on the ground at EGNX waiting for departure and got this lovely thing https://gyazo.com/b446bd9275455aa714961706a46ddda0

This will happen if you click on the map links - we are obligated to preserve those for copyright

**Comment by Benjozork** [comment]
Thanks @Awemeter @BritishAvgeek for testing - UX improvements are gonna be made and we will open a next round of testing soon.

**Comment by donstim** [comment]
Just a comment after looking at the neat screenshots above -- probably premature for this kind of comment, but all the weights shown on the dispatch overview page are incorrect and kind of an odd mix of pounds and kilograms. I assume these are probably all just placeholders at this point?

**Comment by TansyEliot** [comment]
> > 
> > 
> > Just doing some private testing sat on the ground at EGNX waiting for departure and got this lovely thing https://gyazo.com/b446bd9275455aa714961706a46ddda0
> 
> This will happen if you click on the map links - we are obligated to preserve those for copyright
> 

Is FBW Team going to make a map for EFB in the future?

**Comment by Benjozork** [comment]
It's already in there, hence the discussion about the bug

**Comment by Corse-eSports** [comment]
Category: QA Trial
Name: Captain Rafael#0001
Date of testing: 02/02/2021
Version of the sim: 1.12.13.0
PR Tested: #3252
Tier of testing: 1
Changes to observe: EFB is alive

Testing technique:  Testing EFB
Steps you've made to test the Pull Request:

Spawned C&D at LFKJ
Ext Power On, Batt 1 and 2 On - EFB turned on
Clicked through all the pages , clicked on company and putted my simbrief username 
Checked the loadsheet  and the dashboard
Media:
image
image
image


Issues (If Any):  The Airplane MAP worked for 1 minute and soon as I loaded the OFP from simbrief disapeared 
image
Simbrief username is white  as the background is white , making impossible to read 
image
The loadsheet page is not aligned making it little dificult  to read ( the fuel part ) 
image


Overall Rating:  Good but needs improvements 
Conclusions of the testing: Amazing feature , with couple of tweaks we are going to make it  , I love the dashboard 

I noticed the metar font size changes compared to before , also if we can make some time of thing , when we click on the EFB with the mouse to write , that disables the mfs bind shortcuts

**Comment by Awemeter** [comment]
Category: QA Tester
Name: Awemeter#7919 
Date of testing: 2/2/2021
Version of the sim: 1.12.13.0
PR Tested: #3252 
Tier of testing: 2
Changes to observe:
Fix map to show your flight and not a random plane.

Testing technique: Tried out
Steps you've made to test the Pull Request:
1. Spawned C&D at kmsy turned on plane
2. Looked at the map to check if it was showing my plane (was not)
3. Entered a flight number in my mcdu then checked the map to see if it was showing my plane( was not)
4. Restarted the flypad (ext power off and on) then checked the map (it was showing my plane)

Flight notes (When done a full flight):

Media: 
image

Issues (If Any): Map icons still broken, Does not show right plane right away.


Overall Rating: To be improved
Conclusions of the testing: Only observed the map fix because it is the only thing that was changed since my last test. Keep up the good work.

**Comment by Benjozork** [comment]
Thank you everyone for testing.

**Comment by BritishAvgeek** [comment]
Thanks for all your work as always for great features!

(And dealing with efb when for 2 months)

**Pull request merged**